### PR TITLE
Use HELO hostname from environment when provided

### PR DESCRIPTION
**Pull request by btlogy**
As described in #19, Google (and other providers) could require a valid HELO hostname when establishing the SMTP connection.
This PR provides an environment variable to set this hostname when needed.

---

## Code Review Checklist (to be filled out by reviewer)

- [ ] Description accurately reflects what changes are being made.
- [ ] Description explains why the changes are being made (or references an issue containing one).
- [ ] The PR appropriately sized.
- [ ] New code has enough tests. 
- [ ] New code has enough documentation to answer "how do I use it?" and "what does it do?".
- [ ] Existing documentation is up-to-date, if impacted.

### CODE REVIEW

**Comment by btlogy** [comment]
Thus, as described in #5 and LeastAuthority/mw-projects#83: the missing smtp.Hello does not look to be a problem for Google (yet)...
I just wonder if the headers of the received message would not be better processed with a valid hostname...

File: README.md [hunk 1]
```diff
@@ -17,56 +17,98 @@ Simple HTTPS API backend service, which will get data in JSON format and send it
...
+
+```
+SMTP_USERNAME="foo@foobar.in" \
+SMTP_PASSWORD="barbazquux" \
+SMTP_USE_INSECURE_TLS=true \
+./feedback-api \
+-from "no-reply@example.com" \
+-to foo@barbaz.com \
+-smtp-server localhost \
+```
```
Comment (vu3rdd) [inline README.md:66]: Do you also want to add an `SMTP_HELO` usage in the example?

File: README.md [hunk 2]
```diff
@@ -17,56 +17,98 @@ Simple HTTPS API backend service, which will get data in JSON format and send it
...
-You can enable insecure TLS by setting the environment variable `SMTP_USE_INSECURE_TLS` to `"true"`. This can be useful if you want to test with a self-sign certificate server in development env.
+ ```
+ curl \
+--request POST \
+--header "Content-Type: application/json" \
+--data '{"feedback":{"title": "Full Feedback Form","rate": {"type" : "numbers","value" : 10},"questions":[{"question":"q1","answer":"a1"},{"question":"q2","answer":"a2"}]}}' \
+http://localhost:8001/v1/feedback
+```
+
+REM: The JSON content needs to matce the defined template!
```
Comment (vu3rdd) [inline README.md:83]: typo: `matce` -> `match` ?

File: README.md [hunk 3]
```diff
@@ -17,56 +17,98 @@ Simple HTTPS API backend service, which will get data in JSON format and send it
...
+
 ### Requirements
 
 - Docker (20.10.5+)
 - Docker Composer (1.25.0+)
 
 ### Configuration
 
-The relevant environment variables described above can be adapted a local `.env` file:
+This method relies exclusively on environment variables which means both the application paramters and the above described variables need to be defined in a local `.env` file (ignored by Git):
```
Comment (vu3rdd) [inline README.md:98]: minor typo: `paramters` -> `parameters`

**Review by vu3rdd — APPROVED** [review]
Looks good to me. Just left a few very minor comments. Please go ahead and merge.

**Review by vu3rdd — APPROVED** [review]
Looks good to me.

File: README.md (hunk 2 above)
Comment (btlogy) [reply README.md:83]: Thank you @vu3rdd.
BTW: where is the template, so I can add a link to it?

Comment (vu3rdd) [reply README.md:83]: https://github.com/LeastAuthority/feedback-api/blob/main/templates.go

File: README.md (hunk 3 above)
Comment (btlogy) [reply README.md]: Thank you: I've fixed both typos.

File: README.md (hunk 1 above)
Comment (btlogy) [reply README.md:67]: I've added this one too.

**Pull request merged**